### DISCUSSION TITLE
[scripts-audit] Add unit testing for vcpkg scripts

### DISCUSSION THREAD

**Discussion by strega-nil-ms** [microsoft/vcpkg#19191] [Ideas]
See title; suggested by @dg0yt, a lot of our code is in scripts, and it would be really good to actually start testing it.

**Comment by dg0yt** [comment]
Areas of interest:
- commands that modify cmake behaviours, e.g. 
  `find_package` (cf. #18631).
- commands that mimic cmake behaviour, e.g. 
  `vcpkg_list` (cf. #18397).
- all other commands.
